lib/attachProof.js

```diff
@@ -57,15 +59,18 @@ async function attachProofs(operation, {
     // send to ticket service for a proof
     operation = await attachTicketServiceProof({client, ...operation});
   }
-
+  // if there is no invocationTarget defined try to find one
+  if(!invocationTarget) {
+    invocationTarget = _getInvocationTarget({operation});
+  }
   const controller = capabilityInvocationKeyPair.id || signer.id;
   // attach capability invocation proof
-  const capabilityAction = (operation.type === 'CreateWebLedgerRecord') ?
-    'create' : 'write';
-  operation = await attachInvocationProof(operation, {
+  operation = await attachInvocationProof({
+    operation,
     capability: did,
-    capabilityAction,
+    capabilityAction: 'write',
     controller,
+    invocationTarget,
     key: capabilityInvocationKeyPair,
```

Comment:
Why is a `key` being passed in addition to a `signer`? I see that's what was happening before and this PR doesn't change it, but it seems unnecessary (only a signer should be needed). Just noting that.

Reply:
@dmitrizagidulin any idea on this one?

Reply:
If key is passed in, signer is derived from it. I agree that we should just be passing in a signer... a lot of the ?legacy? code tries to pass the key in. There are other issues w/ the design, such as naming this thing a VerificationKey in a lot of places when it's really a keypair... we need to have a redesign discussion after we make the latest releases... I think the serialized structures are fine for now... but the library design needs a bit of work.

Reply:
I made an issue about it: https://github.com/veres-one/did-veres-one/issues/70